src/app.rs

```diff
@@ -51,9 +51,13 @@ pub async fn build_with_cert(
     .await?;
 
     let shutdown_trigger = shutdown.trigger();
-    let admin = admin::Builder::new(config.clone(), workload_manager.workloads(), ready)
+    let admin = admin::Builder::new(config.clone(), workload_manager.workloads())
         .bind(registry, shutdown_trigger)
         .expect("admin server starts");
+    let readiness_server = readiness::Builder::new(config.clone(), ready)
+        .bind(shutdown.trigger())
+        .expect("error starting readiness server");
```

Comment:
I know we did it before but 
```suggestion
        .context("error starting readiness server")?;
```

would give better error messages